MoltenVK/MoltenVK/GPUObjects/MVKBuffer.mm

```diff
@@ -187,22 +187,21 @@
 																   offset: _deviceMemoryOffset];	// retained
 			propagateDebugName();
 			return _mtlBuffer;
-#if MVK_MACOS
-		} else if (_isHostCoherentTexelBuffer) {
-			// According to the Vulkan spec, buffers, like linear images, can always use host-coherent memory.
-			// But texel buffers on Mac cannot use shared memory. So we need to use host-cached memory here.
-			_mtlBuffer = [_device->getMTLDevice() newBufferWithLength: getByteCount()
-															  options: MTLResourceStorageModeManaged];	// retained
-			propagateDebugName();
-			return _mtlBuffer;
-#endif
 		} else {
 			return _deviceMemory->getMTLBuffer();
 		}
 	}
 	return nil;
 }
 
+id<MTLBuffer> MVKBuffer::getMTLBufferCache() {
+    if (_isHostCoherentTexelBuffer && !_mtlBufferCache) {
+        _mtlBufferCache = [_device->getMTLDevice() newBufferWithLength: getByteCount()
+                                                               options: MTLResourceStorageModeManaged];    // retained
```

Comment:
`MTLResourceStorageModeManaged` breaks iOS & tvOS builds. Needs to be wrapped in `#if MVK_MACOS` conditional build.